### DISCUSSION TITLE
Working with different versions

### DISCUSSION THREAD

**Discussion by hziya** [debauchee/barrier#1968] [Q&A]
I have an ubuntu 20.04 and macos monterey 12.6.3. There is 2.3.2 version for Ubuntu 20.04 and 2.4.0 for monterey. 

Can i connect these two? When i start one of them as server and other as client. client is stuck in "Barrier is starting" phase. 

Should I compile 2.4.0 in ubuntu?

Thank you in advance

**Comment by DeepYogurt** [comment]
I use Barrier 2.4.0 as server on macOS Mojave and Barrier 2.3.3 as client on another macOS High Sierra and it works most of the time, but with a couple of funny things. It occasionally does not connect with the "Barrier is starting" but not very often - I just stop and restart both server and client and it gets connected usually. The biggest problem is that there are some windows that mouse clicks do not work with but have not figured out what the common thread is for the ones that don't - but most work fine when it does connect to server. Did find a bug with the mouse scroll wheel not working at all but only if I set the server side Mouse System Settings "Scrolling Speed" slider to "Fast" (the right-most position of the slider". Setting it to just one position less fixed the problem.

Good luck...

**Reply by DeepYogurt** [reply]
I learned what the problem was with Barrier not working with some windows - specifically Little Snitch notification windows. You need to change Little Snitch Preferences - click the "Security & Privacy" tab and then click the padlock at the bottom and enter your login password to change something. Then put a checkmark in the "Allow GUI Scripting access to Little Snitch" option and that should do it. Apparently Little Snitch considers this a possible security issue in order to prevent remote intrusions. I'm using Barrier on my own local network so I don't see this as a very likely threat. But use your own opinion as to whether you wish to allow this.

There are still some issues/bugs with Barrier, but it is definately an improvement over using a desk full of keyboards and mice. The biggest issue is you still need a local keyboard and mouse until Barrier is running - like during single-user startup mode or recovery mode or safe mode or any mode that requires special keyboard commands typed in to invoke. Am using it with four Mac systems and it is just such an intuitive user interface with moving the cursor to the screen edge and then over into the adjacent computer.

Kudo's to all the folks who had a hand in getting this working!!!
